build-in-container/scripts/build.sh

```diff
@@ -0,0 +1,207 @@
+# This overwrites packaged logs from previous calls to this function
+# The logs from previous calls are preserved and packaged as long as
+#  `make clean` has not been called between builds of separate repos
+#
+# No arguments
+# Global variables expected to be defined: LOG_PUBLISH_DIR, BUILD_DIR
+publish_build_logs() {
+    log "-- pack logs"
+    mkdir -p "$LOG_PUBLISH_DIR"
+    if [[ -d $BUILD_DIR/logs ]]; then
+        tar -C "$BUILD_DIR/logs" -czf "$LOG_PUBLISH_DIR/pkggen.logs.tar.gz" .
+    else
+        echo "-- Warning - no 'logs' folder under $BUILD_DIR"
+    fi
+    log "-- pack package build artifacts"
+    if [[ -d $BUILD_DIR/pkg_artifacts ]]; then
+        tar -C "$BUILD_DIR/pkg_artifacts" -czf "$LOG_PUBLISH_DIR/pkg_artifacts.tar.gz" .
+    else
+        echo "-- Warning - no 'pkg_artifacts' folder under $BUILD_DIR"
+    fi
+}
+
+#
+# main
+#
+
+trap cleanup EXIT
+
+echo "============================"
+echo "Building overlake-packages  "
+echo "============================"
+
+# parse script parameters:
+#
+# -a -> build artifacts folder name
+# -c -> chroot base dir
+# -o -> root folder for out and build dir
+# -p -> folder where built RPMs/SRPMs should be published
+# -n -> nb of chroots (this will be used to parallelize the build, 0 means as many as machine supports)
+# -d -> enable development mode (ccache, etc).
+
+while getopts ":a:c:o:p:q:n:dv" OPTIONS; do
+  case "${OPTIONS}" in
+    a ) BUILD_ARTIFACTS_FOLDER_NAME=$OPTARG ;;
+    c ) CHROOT_BASE_DIR=$OPTARG ;;
+    o ) BUILD_OUT_BASE_DIR=$OPTARG
+        BUILD_DIR=$BUILD_OUT_BASE_DIR/build
+        CCACHE_DIR=$BUILD_OUT_BASE_DIR/ccache
+        OUT_DIR=$BUILD_OUT_BASE_DIR/out ;;
+    p ) ARTIFACT_PUBLISH_DIR=$OPTARG ;;
+    q ) LOG_PUBLISH_DIR=$OPTARG ;;
+    n ) CHROOT_NB=$OPTARG ;;
+    d ) USE_CCACHE=y ;;
+    v ) VERBOSE=1
+        if [[ "$LOG_LEVEL" == "info" ]]; then
+            LOG_LEVEL="debug"
+        else
+            LOG_LEVEL="info"
+        fi
+        set -x ;;
+    \? )
+        echo "Error - Invalid Option: -$OPTARG" 1>&2
+        exit 1
+        ;;
+    : )
+        echo "Error - Invalid Option: -$OPTARG requires an argument" 1>&2
+        exit 1
+        ;;
+  esac
+done
+
+echo "-- BUILD_ARTIFACTS_FOLDER_NAME        -> $BUILD_ARTIFACTS_FOLDER_NAME"
+echo "-- BUILD_OUT_BASE_DIR                 -> $BUILD_OUT_BASE_DIR"
+echo "-- BUILD_DIR                          -> $BUILD_DIR"
+echo "-- OUT_DIR                            -> $OUT_DIR"
+echo "-- CHROOT_BASE_DIR                    -> $CHROOT_BASE_DIR"
+echo "-- CHROOT_NB (0 = max)                -> $CHROOT_NB"
+echo "-- ARTIFACT_PUBLISH_DIR               -> $ARTIFACT_PUBLISH_DIR"
+echo "-- LOG_PUBLISH_DIR                    -> $LOG_PUBLISH_DIR"
+echo "-- USE_CCACHE                         -> $USE_CCACHE"
+echo ""
+
+download_mariner_toolkit
+
+if [[ "${USE_CCACHE}"=="y" ]]; then
+    log "-- Apply workaround for ccache"
+    # WORKAROUND: The Mariner toolkit is having trouble downloading the ccache
```

Comment:
We should probably figure this out if we're going to capture it in a tutorial?

Reply:
Removing this, will work on ccache in the next revision